pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -283,6 +282,20 @@ func setReservation(disk *api.Disk) {
 	}
 }
 
+func setErrorPolicy(diskDevice *v1.Disk, disk *api.Disk) error {
+	if diskDevice.ErrorPolicy == nil {
+		disk.Driver.ErrorPolicy = v1.DiskErrorPolicyStop
+		return nil
+	}
+	switch *diskDevice.ErrorPolicy {
+	case v1.DiskErrorPolicyStop, v1.DiskErrorPolicyIgnore, v1.DiskErrorPolicyReport, v1.DiskErrorPolicyEnospace:
+		disk.Driver.ErrorPolicy = *diskDevice.ErrorPolicy
+	default:
+		return fmt.Errorf("error policy %s not recognized", *diskDevice.ErrorPolicy)
+	}
+	return nil
+}
```

Comment:
A couple of questions here:
- Now this does not accept an empty string `""` as a valid policy and this function will return an error. Is it intentional? Or we just miss here an additional check? Note: just asking :)
- Should we have a validation webhook checking that the VMI yaml has correct policy?

Reply:
> A couple of questions here:
> 
> * Now this does not accept an empty string `""` as a valid policy and this function will return an error. Is it intentional? Or we just miss here an additional check? Note: just asking :)

I changed the [ErrorPolicy](https://github.com/kubevirt/kubevirt/blob/256c29b991667ba98bfb585c77f09a193fd99ad9/staging/src/kubevirt.io/api/core/v1/schema.go#L620) to a pointer if the field isn't defined in the VMI specification then `diskDevice.ErrorPolicy` will be `nil`. This unit test should cover this [case](https://github.com/kubevirt/kubevirt/blob/256c29b991667ba98bfb585c77f09a193fd99ad9/pkg/virt-launcher/virtwrap/converter/converter_test.go#L2052). Did I get correctly your point?

> * Should we have a validation webhook checking that the VMI yaml has correct policy?

I don't want to force the users to set this field. If not set then we should fall back to the default behavior to pause the VM on IOerrors

Reply:
> I changed the [ErrorPolicy](https://github.com/kubevirt/kubevirt/blob/256c29b991667ba98bfb585c77f09a193fd99ad9/staging/src/kubevirt.io/api/core/v1/schema.go#L620) to a pointer if the field isn't defined in the VMI specification then diskDevice.ErrorPolicy will be nil. This unit test should cover this [case](https://github.com/kubevirt/kubevirt/blob/256c29b991667ba98bfb585c77f09a193fd99ad9/pkg/virt-launcher/virtwrap/converter/converter_test.go#L2052). Did I get correctly your point?

Right, its now a pointer, but I was rather wondering here what should happen if a user specified an empty value `""` for the policy in the VMI spec:
```yaml
            errorPolicy: ""
```
Currently, this will lead to `error policy  not recognized`. But maybe we should instead fall back to the default `v1.DiskErrorPolicyStop` (same as with an empty value)?... I personally do not have a strong opinion on that. Just wanted to ensure we cover that aspect as well.

> I don't want to force the users to set this field. If not set then we should fall back to the default behavior to pause the VM on IOerrors

Yes, no need to force the field. However, currently nothing prevents me from setting a wrong value in the VMI yaml:
```yaml
            errorPolicy: "WrongPolicy"
```
and I will get the error only on VM sync when conversion takes place. I think it would make sense to **not** admit such a VMI spec, i.e. to add the validation in the admission web hook: if the field is not `nil`, then check the values.

Reply:
I see, let me check what we do for `cache`. I think this should follow the same behavior

Reply:
You are right, there is a missing part, for cache we do something like this: https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go#L2478-L2485